### PR TITLE
[BEAM-9299] Upgrade Flink Runner from 1.8.2 to 1.8.3

### PR DESCRIPTION
**Pull request by sunjincheng121**
Upgrade Flink Runner 1.8.x to 1.8.3

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by sunjincheng121** [comment]
Run Java PreCommit

**Comment by angoenka** [comment]
Unfortunately, jenkins run the tests against latest Flink supported by beam (1.9) 
Reference https://github.com/apache/beam/blob/cec1094adba3dd20f382fc07409fbf7fb58fbbc6/sdks/python/test-suites/portable/common.gradle#L57 

Can you please test it separately to make sure the version update passes the tests.

**Comment by sunjincheng121** [comment]
Hi @angoenka Thanks for your comment, I found that the issue FLINK-15844 also exists in Flink 1.8.3. I noticed that FLINK-15844 has provided a fix for 1.9.3. However, AFAIK, there will be no new 1.8 releases any more according to Flink's release policy]. There are two solutions I have left in the JIRA. I appreciate if you have a look at it :)

**Comment by angoenka** [comment]
Thanks for summarizing the options. I added my comment in the jira.

**Comment by sunjincheng121** [comment]
As BEAM-5396 and FLINK-11048 have been completed so far, we have a cleaner solution 3, which is to remove `BeamFlinkRemoteStreamEnvironment` class. I have updated the PR, but the content does not include the changes of `cover all of the version for test` which [~angoenka] mentioned, I think we can handle it in another PR.  

What do you think?

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java [hunk 1]
```diff
@@ -311,78 +306,4 @@ private static void applyLatencyTrackingInterval(
...
   }
-
-  /**
-   * Remote stream environment that supports job execution with restore from savepoint.
-   *
-   * <p>This class can be removed once Flink provides this functionality.
-   *
-   * <p>TODO: https://issues.apache.org/jira/browse/BEAM-5396
-   */
-  private static class BeamFlinkRemoteStreamEnvironment extends RemoteStreamEnvironment {
```
Comment (mxm) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java]: We cannot remove this until we remove 1.7 support.

**Comment by mxm** [comment]
Could you explain what the issue is here? The upgrade from 1.8.2 to 1.8.3 should not require any code changes.

**Comment by sunjincheng121** [comment]
Hi @mxm ，thanks for the review, I have update the PR, appreciate if you have a look at the changes :)

**Pull request merged**